devito/operator.py

```diff
@@ -109,32 +108,81 @@ def __init__(self, expressions, **kwargs):
         self.dle_flags = dle_state.flags
         self.func_table.update(OrderedDict([(i.name, MetaCall(i, True))
                                             for i in dle_state.elemental_functions]))
-        parameters.extend([i.argument for i in self.dle_arguments])
         self.dimensions.extend([i.argument for i in self.dle_arguments
                                 if isinstance(i.argument, Dimension)])
         self._includes.extend(list(dle_state.includes))
 
         # Introduce the required symbol declarations
         nodes = iet_insert_C_decls(dle_state.nodes, self.func_table)
 
-        # Initialise ArgumentEngine
-        self.argument_engine = ArgumentEngine(clusters.ispace, parameters,
-                                              self.dle_arguments)
+        # Insert data and pointer casts for array parameters and profiling structs
+        casts = [ArrayCast(f) for f in self.input if f.is_Tensor and f._mem_external]
```

Comment:
I'm re-adding a comment that GitHub dropped after the latest changes: this should better go in ``ir/iet/scheduler`` as a ``iet_insert_casts`` free function